nncf/quantization/quantization_metric.py

```diff
@@ -0,0 +1,273 @@
+            insertion_data = prop_graph_solver.run_on_ip_graph(insertion_point_graph)
+            return len(insertion_data)
+
+    def collect(self):
+        for quantizer in self._all_quantizations.values():
+            self.params.add(quantizer.num_bits)
+
+        for h in self.header:
+            self.stat[h] = {}
+            for p in self.params:
+                self.stat[h][p] = 0
+
+        for quantizer in self._all_quantizations.values():  # type: BaseQuantizer
+            num_bits = quantizer.num_bits
+            self.stat[self.TOTAL_RATIO_STR][num_bits] += 1
+            self.stat[self.TOTAL_RATIO_STR][self.IN_ALL_STR] += 1
+            type_ = self.WEIGHTS_RATIO_STR if quantizer.is_weights else self.ACTIVATIONS_RATIO_STR
+            self.stat[type_][num_bits] += 1
+            self.stat[type_][self.IN_ALL_STR] += 1
+            if quantizer.per_channel:
+                self.stat[type_][self.PER_CHANNEL_STR] += 1
+                self.stat[self.TOTAL_RATIO_STR][self.PER_CHANNEL_STR] += 1
+            if quantizer.signed:
+                self.stat[type_][self.SIGNED_STR] += 1
+                self.stat[self.TOTAL_RATIO_STR][self.SIGNED_STR] += 1
+            if isinstance(quantizer, SymmetricQuantizer):
+                self.stat[type_][self.SYMMETRIC_STR] += 1
+                self.stat[self.TOTAL_RATIO_STR][self.SYMMETRIC_STR] += 1
+            else:
+                self.stat[type_][self.ASYMMETRIC_STR] += 1
+                self.stat[self.TOTAL_RATIO_STR][self.ASYMMETRIC_STR] += 1
+        self.stat[self.TOTAL_WEIGHTS_RATIO_STR] = deepcopy(self.stat[self.WEIGHTS_RATIO_STR])
+        self.stat[self.TOTAL_ACTIVATIONS_RATIO_STR] = deepcopy(self.stat[self.ACTIVATIONS_RATIO_STR])
+        for h in self.header[1:]:
+            for key, value in self.stat[h].items():
+                self.stat[h][key] /= self._ratio[h]
+                self.stat[h][key] *= 100
+
+    def get_metric_table(self):
+        table = Texttable()
+        header_for_table = [h + '(={})'.format(self._ratio[h]) for h in self.header[1:]]
+        
+        data = [self.header]
+        for p in self.params:
+            drow = {h: 0 for h in self.header}
+            for column_name in self.header:
+                drow[column_name] = self.stat[column_name][p]
+            drow[self.PARAMS_STR] = p
+            row = [drow[h] for h in self.header]
+            data.append(row)
+        data[0] =[self.PARAMS_STR] + header_for_table
+        table.add_rows(data)
+        return table
+
+class MemoryСostMetric(BaseMetric):
+    """
+
+    This metric considers how many times memory consumption for network weights will decrease.
+    
+    """
+    PARAMS_STR = 'params'
+    NAME_STR = 'MemoryCost'
+
+    EXPECTED_MEMORY_CONSUMPTION_DECREASE_STR = 'memory consumption decrease'
+    SIZE_MEMORY_FP_WEIGHTS_STR = 'size memory for fp weights (bit)'
+    SIZE_MEMORY_COMPRESSED_WEIGHTS_STR = 'size memory for quantized weights (bit)'
```

Comment:
Still unprofessional. Have you seen a table anywhere in a production-grade document with headers that don't at least start with a capital letter?

Bits seem to be too small a unit for memory consumption, wouldn't you agree?